### PR TITLE
Case and person anonymization - Anonymization UI

### PR DESCRIPTION
**Pull request by atoncetti**

### CODE REVIEW

**Comment by maennchen** [comment]
@atoncetti Please use rebases instead of merges. Otherwise, I will have a hard time reviewing this.

* `git fetch && git rebase -i origin/feat/case_and_person_anonymization`
* remove all commits that were not done by you
* solve conflicts if there are any
* push with `--force-with-lease`

**Comment by maennchen** [comment]
* Reload Header when redacting / unredacting.
* When disabling redaction/reidentification buttons, put a reason next. (popover)
* List view case / person add redacted filter
* Update on the person/case chooser
  - chooser in create cases: disable selection of redacted entries
  - chooser in inbox: allow choosing redacted

File: lib/hygeia/auto_tracing_context/auto_tracing.ex [hunk 1]
```diff
@@ -209,6 +209,8 @@ defmodule Hygeia.AutoTracingContext.AutoTracing do
             user :: :anonymous | User.t() | Person.t(),
             meta :: %{atom() => term}
           ) :: boolean
+    def authorized?(_auto_tracing, _action, _user, %{case: %Case{redacted: true}}), do: false
```
Comment (janmrnak) [inline lib/hygeia/auto_tracing_context/auto_tracing.ex:212]: Is this the correct place?

File: lib/hygeia_web/live/case_live/choose.ex [hunk 2]
```diff
@@ -74,15 +76,18 @@ defmodule HygeiaWeb.CaseLive.Choose do
...
       )
+      |> maybe_discard_anonymized(socket.assigns.discard_anonymized)
+      |> Repo.all()
 
     assign(socket, cases: Repo.preload(cases, person: [tenant: []]))
   end
 
   defp load_case(uuid), do: uuid |> CaseContext.get_case!() |> Repo.preload(person: [tenant: []])
+
+  defp maybe_discard_anonymized(query, true), do: where(query, [case], case.redacted)
```
Comment (maennchen) [inline lib/hygeia_web/live/case_live/choose.ex:91]:
```suggestion
  defp maybe_discard_anonymized(query, true), do: where(query, [case], not case.redacted)
```

File: lib/hygeia_web/live/case_live/index.sface [hunk 3]
```diff
@@ -144,6 +144,31 @@
...
                     </Field>
                   </div>
+                  <div class="form-group">
+                    <label class="d-block">{gettext("Show cases")}</label>
+                    <Field name="anonymization" class="form-check form-check-inline">
+                      <label class="form-check-label radio-label">
+                        <RadioButton
+                          class="form-check-input"
+                          value="any"
+                          checked={match?("any", form.source["anonymization"])}
```
Comment (maennchen) [inline lib/hygeia_web/live/case_live/index.sface:154]:
```suggestion
                          checked={form.source["anonymization"] == "any"}
```

Comment (maennchen) [reply lib/hygeia_web/live/case_live/index.sface:154]: needed?

File: lib/hygeia_web/live/case_live/transmissions.sface [hunk 4]
```diff
@@ -36,7 +36,7 @@
           <h3>
             {gettext("Propagated")}
           </h3>
-          <div :if={authorized?(Case, :create, auth, tenant: :any)}>
+          <div :if={authorized?(Transmission, :create, auth, tenant: :any, case: @case)}>
```
Comment (maennchen) [inline lib/hygeia_web/live/case_live/transmissions.sface:39]: revert

File: lib/hygeia_web/live/person_live/base_data.sface [hunk 5]
```diff
@@ -57,14 +65,51 @@
             person_uuid: @person,
             return_to: Routes.case_index_path(@socket, :index)
           )}
-          :if={authorized?(Case, :create, auth, tenant: :any)}
+          :if={authorized?(Case, :create, auth, tenant: :any) and not @person.redacted}
```
Comment (maennchen) [inline lib/hygeia_web/live/person_live/base_data.sface:68]:
```suggestion
          :if={authorized?(Case, :create, auth, tenant: :any, person: @person)}
```

check => auth

File: lib/hygeia_web/live/person_live/choose.ex [hunk 6]
```diff
@@ -87,4 +88,7 @@ defmodule HygeiaWeb.PersonLive.Choose do
 
   defp format_date(nil), do: nil
   defp format_date(date), do: HygeiaCldr.Date.to_string!(date)
+
+  defp maybe_discard_anonymized(query, true), do: where(query, [person], person.redacted)
```
Comment (maennchen) [inline lib/hygeia_web/live/person_live/choose.ex:92]:
```suggestion
  defp maybe_discard_anonymized(query, true), do: where(query, [person], not person.redacted)
```

File: lib/hygeia_web/live/person_live/index.sface [hunk 7]
```diff
@@ -116,6 +116,32 @@
...
               </div>
+
+              <div class="form-group">
+                <label class="d-block">{gettext("Show people")}</label>
+                <Field name="anonymization" class="form-check form-check-inline">
+                  <label class="form-check-label radio-label">
+                    <RadioButton
+                      class="form-check-input"
+                      value="any"
+                      checked={match?("any", form.source["anonymization"])}
```
Comment (maennchen) [inline lib/hygeia_web/live/person_live/index.sface:127]: needed?

File: priv/gettext/Person Base Data.pot [hunk 8]
```diff
@@ -0,0 +1,11 @@
+## This file is a PO Template file.
```
Comment (maennchen) [inline priv/gettext/Person Base Data.pot:1]: Wrong

File: priv/gettext/de/LC_MESSAGES/default.po [hunk 9]
```diff
@@ -3742,6 +3771,10 @@ msgstr "Person ist symptomatisch und wird darum getestet"
 msgid "Person living in the same household or in a very close relationship. (e.g. couple that is not living together)"
 msgstr "Personen vom selben Haushalt oder mit sehr engem Kontakt. ( z.B. Paar, das nicht zusammen lebt)"
 
+#, elixir-autogen, icu-format
+msgid "Person redacted successfully"
+msgstr "Person erfolgreich redigiert"
```
Comment (maennchen) [inline priv/gettext/de/LC_MESSAGES/default.po:3776]:
```suggestion
msgstr "Person erfolgreich anonymisiert"
```

File: lib/hygeia/auto_tracing_context/auto_tracing.ex (hunk 1 above)
Comment (atoncetti) [reply lib/hygeia/auto_tracing_context/auto_tracing.ex:212]: I would argue it is, since we want to match redacted cases with quite high priority

**Pull request merged**